static/js/popovers.js

```diff
@@ -629,17 +629,27 @@ exports.user_info_popover_handle_keyboard = function (key) {
     popover_items_handle_keyboard(key, items);
 };
 
-exports.show_sender_info = function () {
-    var $message = $(".selected_message");
-    var $sender = $message.find(".sender_info_hover");
-    var $prev_message = $message.prev();
-    while (!$sender[0]) {
-        $prev_message = $prev_message.prev();
-        if (!$prev_message) {
-            break;
+exports.prev_with_avatar = function (row) {
+    var prev_row;
+
+    while (row.find('.inline_profile_picture').length === 0) {
+        prev_row = row.prev();
+
+        if (prev_row.length === 0) {
+            blueslip.error('Could not find row with profile picture');
+            return row;
         }
-        $sender = $prev_message.find(".sender_info_hover");
+
+        row = prev_row;
     }
```

Comment:
I realize that this now does what the code here was clearly trying to do, but I think this change is effectively a UX regression in the case that the message is offscreen (you end up with this):

![image](https://user-images.githubusercontent.com/2746074/53984487-44030500-40ce-11e9-9269-c91ae5fec69c.png)

There's also definitely a bug starting with this commit where the currently selected message jumps away from the one you had selected, and instead becomes a potentially offscreen one, which breaks keyboard navigation.

I think we might want to solve this by just always popping next to the current message, and get rid of this (apparently broken) loop.

@rishig FYI for confirmation on the product question.

Reply:
I'm a little confused on what's going on here. Is it that there's a long /me message, and clicking the avatar causes the message to jump up?

(Maybe a tangent, but I kind of think we should just disallow multi-line /me messages.)

Reply:
No, it's if you have a series of 15 messages by the same sender, and you hit the `u` hotkey, do we pop the profile popover next to the message you have selected vs. next to the avatar (several messages up).

This commit isn't /me related directly.

Reply:
ah, got it. I think "next to message you have selected" makes sense, rather than changing the selected message (so agreed with what you said).